webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/controllers/VolumeTracingZarrStreamingController.scala

```diff
@@ -0,0 +1,256 @@
+import com.scalableminds.util.mvc.ExtendedController
+import com.scalableminds.util.tools.{Fox, FoxImplicits}
+import com.scalableminds.webknossos.datastore.VolumeTracing.VolumeTracing
+import com.scalableminds.webknossos.datastore.dataformats.MagLocator
+import com.scalableminds.webknossos.datastore.dataformats.zarr.{ZarrCoordinatesParser, ZarrSegmentationLayer}
+import com.scalableminds.webknossos.datastore.datareaders.jzarr.{OmeNgffGroupHeader, OmeNgffHeader, ZarrHeader}
+import com.scalableminds.webknossos.datastore.datareaders.{ArrayOrder, AxisOrder}
+import com.scalableminds.webknossos.datastore.helpers.ProtoGeometryImplicits
+import com.scalableminds.webknossos.datastore.models.WebKnossosDataRequest
+import com.scalableminds.webknossos.datastore.models.datasource.{DataLayer, ElementClass}
+import com.scalableminds.webknossos.datastore.services.UserAccessRequest
+import com.scalableminds.webknossos.tracingstore.tracings.editablemapping.{EditableMappingService, RemoteFallbackLayer}
+import com.scalableminds.webknossos.tracingstore.tracings.volume.VolumeTracingService
+import com.scalableminds.webknossos.tracingstore.{
+  TSRemoteDatastoreClient,
+  TSRemoteWebKnossosClient,
+  TracingStoreAccessTokenService
+}
+import play.api.i18n.Messages
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class VolumeTracingZarrStreamingController @Inject()(
+    tracingService: VolumeTracingService,
+    accessTokenService: TracingStoreAccessTokenService,
+    editableMappingService: EditableMappingService,
+    remoteDataStoreClient: TSRemoteDatastoreClient,
+    remoteWebKnossosClient: TSRemoteWebKnossosClient)(implicit ec: ExecutionContext)
+    extends ExtendedController
+    with ProtoGeometryImplicits
+    with FoxImplicits {
+
+  override def defaultErrorCode: Int = NOT_FOUND
+
+  def volumeTracingFolderContent(token: Option[String], tracingId: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      accessTokenService.validateAccess(UserAccessRequest.readTracing(tracingId), urlOrHeaderToken(token, request)) {
+        for {
+          tracing <- tracingService.find(tracingId) ?~> Messages("tracing.notFound") ~> NOT_FOUND
+          existingMags = tracing.resolutions.map(vec3IntFromProto)
+        } yield
+          Ok(
+            views.html.datastoreZarrDatasourceDir(
+              "Tracingstore",
+              "%s".format(tracingId),
+              List(".zattrs", ".zgroup") ++ existingMags.map(_.toMagLiteral(allowScalar = true))
+            )).withHeaders()
+      }
+    }
+
+  def volumeTracingFolderContentJson(token: Option[String], tracingId: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      accessTokenService.validateAccess(UserAccessRequest.readTracing(tracingId), urlOrHeaderToken(token, request)) {
+        for {
+          tracing <- tracingService.find(tracingId) ?~> Messages("tracing.notFound") ~> 404
```

Comment:
Can you also use `NOT_FOUND` here?
```suggestion
          tracing <- tracingService.find(tracingId) ?~> Messages("tracing.notFound") ~> NOT_FOUND
```